commit b12b24fb433bd835944544349e6cf85575293777
Author: Alexandre BIQUE <bique.alexandre@gmail.com>
Date:   Fri Nov 5 10:10:30 2021 +0100

    Revert "[RuntimeDyld][COFF] Report fatal error on error, rather than emiting diagnostic."
    
    This reverts commit b45b5166f8f91e601ebd7bc73bb6800155c4c9bc.

diff --git a/llvm/lib/ExecutionEngine/RuntimeDyld/Targets/RuntimeDyldCOFFX86_64.h b/llvm/lib/ExecutionEngine/RuntimeDyld/Targets/RuntimeDyldCOFFX86_64.h
index 9df3e2e3c3bf..ebe3ca33d308 100644
--- a/llvm/lib/ExecutionEngine/RuntimeDyld/Targets/RuntimeDyldCOFFX86_64.h
+++ b/llvm/lib/ExecutionEngine/RuntimeDyld/Targets/RuntimeDyldCOFFX86_64.h
@@ -113,10 +113,11 @@ public:
       // The MemoryManager can make sure this is always true by forcing the
       // memory layout to be: CodeSection < ReadOnlySection < ReadWriteSection.
       const uint64_t ImageBase = getImageBase();
-      if (Value < ImageBase || ((Value - ImageBase) > UINT32_MAX))
-        report_fatal_error("IMAGE_REL_AMD64_ADDR32NB relocation requires an "
-                           "ordered section layout");
-      else {
+      if (Value < ImageBase || ((Value - ImageBase) > UINT32_MAX)) {
+        llvm::errs() << "IMAGE_REL_AMD64_ADDR32NB relocation requires an"
+                     << "ordered section layout.\n";
+        write32BitOffset(Target, 0, 0);
+      } else {
         write32BitOffset(Target, RE.Addend, Value - ImageBase);
       }
       break;
